core/integration/module_java_test.go

```diff
@@ -288,6 +288,84 @@ func (JavaSuite) TestConstructor(_ context.Context, t *testctx.T) {
 	})
 }
 
+func (JavaSuite) TestEnum(_ context.Context, t *testctx.T) {
+	t.Run("can use an enum value", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("print", "--severity=LOW")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "LOW", out)
+	})
+
+	t.Run("can not use a value not defined in the enum", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		_, err := javaModule(t, c, "enums").
+			With(daggerCall("print", "--severity=FOO")).
+			Stdout(ctx)
+
+		require.Error(t, err)
+	})
+
+	t.Run("can return an enum value", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("from-string", "--severity=MEDIUM")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "MEDIUM", out)
+	})
+
+	t.Run("can return a list of enum values", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("get-severities-list")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "LOW\nMEDIUM\nHIGH\n", out)
+	})
+
+	t.Run("can return an array of enum values", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("get-severities-array")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "LOW\nMEDIUM\nHIGH\n", out)
+	})
+
+	t.Run("can read list of enum values", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("list-to-string", "--severities=MEDIUM,LOW")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "MEDIUM,LOW", out)
+	})
+
+	t.Run("can read array of enum values", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		out, err := javaModule(t, c, "enums").
+			With(daggerCall("array-to-string", "--severities=HIGH,LOW")).
+			Stdout(ctx)
+
+		require.NoError(t, err)
+		require.Equal(t, "HIGH,LOW", out)
+	})
+}
+
 func javaModule(t *testctx.T, c *dagger.Client, moduleName string) *dagger.Container {
```

Comment:
Just a quick note but I really like this pattern for testing features of an SDK, instead of inlining the source code like we do in others.
@helderco do you think we could migrate other SDKs to that pattern when we have some time?
This could be an opportunity to refactor tests and simplify/optimize them

It's also a very nice way to play with a standalone test without having to run the whole dev engine when something is not working properly. It's something I do very often when I work on a SDK.
Good idea 🚀 ⭐

Reply:
We started out like that in Go, but adding Python and then TypeScript it was way harder to have parametrized tests with per-sdk sources because you have to open different files to compare. Then there's the extra boilerplate and files if every module in tests come from the file system. Making them inline and generated on demand made it much easier to maintain. Still, we do sometimes commit a `dagger init`ed module under `core/integration/modules` when it makes sense or makes things easier.

Reply:
I was more interested for specific SDK tests, not for common integrations tests that tests multiples SDK, I didn't remember seeing that pattern except for interfaces.
Anyway thanks for the explanations :)